app/indexer/settling/BastionIndexer.ts

```diff
@@ -16,36 +16,71 @@ export default class BastionIndexer extends BaseContractIndexer {
   }
 
   async bastionArmyMoved(event: Event): Promise<void> {
+    const eventId = event.eventId as string;
     const params = event.parameters;
-    const lon = BigNumber.from(params[0]).toNumber();
-    const lat = BigNumber.from(params[1]).toNumber();
+    const longitude = BigNumber.from(params[0]).toNumber();
+    const latitude = BigNumber.from(params[1]).toNumber();
     const bastionPastLocation = parseInt(params[2]);
     const bastionCurrentLocation = parseInt(params[3]);
-    const realmId = arrayUInt256ToNumber(params.slice(4, 6));
-    const armyId = parseInt(params[6]);
-    const bastionId = createBastionId(lat, lon);
+    const bastionArrivalBlock = parseInt(params[4]);
+    const realmId = arrayUInt256ToNumber(params.slice(5, 7));
+    const armyId = parseInt(params[7]);
+    const bastionId = createBastionId(latitude, longitude);
 
+    // only index events with 8 params
+    // 1 param was added in event in the last version of the contract
+    if (params.length === 8) {
     try {
-      await this.context.prisma.army.update({
+      const promises = await Promise.allSettled([
+      this.context.prisma.army.update({
         where: { realmId_armyId: { realmId, armyId } },
         data: {
           bastionId,
           bastionPastLocation,
           bastionCurrentLocation,
-          bastionArrivalBlock: event.blockNumber,
+          bastionArrivalBlock: bastionArrivalBlock,
+        },
+      }),
+      this.saveRealmHistory({
+        realmId,
+        bastionId: bastionId,
+        eventId,
+        eventType: "bastion_army_move",
+        timestamp: event.timestamp,
+        transactionHash: event.txHash,
+        data: {
+          armyId,
+          bastionPastLocation,
+          bastionCurrentLocation,
+          bastionArrivalBlock,
         },
-      });
-    } catch (e) {}
+      })
+    ]);
+    console.log(promises)
```

Comment:
can remove log, as well as the promises declaration

Reply:
yes sorry for that